_posts/2017-03-04-bettracker-typescript.md

```diff
@@ -0,0 +1,108 @@
+
+Process wise was a little more tricky, we were about to release a change which made changes to pretty much every file in a codebase which was being worked on daily. In order to avoid a constant loop of merge conflicts we froze any other development on the codebase for 2-3 days whilst we initially brought our feature branch up to date with master (resolving many conflicts as we went!) and then had the opportunity to release and test the refactor without having to worry about new code being released behind it and complicating the whole process.
+
+Fortunately after a full regression test of our product, the release went to plan, the only bug we encountered was a display issue with an icon tag, which probably came about as a result of bringing our branch up to date with master - this was seen as a success given the scale of the diff that we'd just released.
+
+
+# Result
+Our My Bets product is now fully TypeScripted. We've not added a type hint to every parameter or method but in order to address our initial problem, we've added types to all of our models for bets, selections and outcomes.
+
+Instead of looking at the codebase as a whole, we're going to add types to parts of the code as we're working on it. For example if I'm working on a component such as a scoreboard, displaying live football scores whilst the bet is in play, I might go into that component, think about all the props and then add in the relevant types as part of my current ticket.
+
+We've found the more types we add in, the more support we get from our IDE, which is a good thing, we can click around between classes easier because the IDE can infer more from our code, this is good as it makes development easier and gives us confidence in what data structures we can expect in certain areas of our code.
+
+Our development process is still much the same as it was before. From a development point of view, the developer can either have `gulp watch` running in the background or just run a build job every time they've done some work. The commands the developer uses with TypeScript are exactly the same as they were when we used babel.
+
+
+# What we've learnt
+We've come a long way, what was initially a project to add a bit more clarity to our models turned into a full scale refactor of our My Bets product. Whilst the work done has taken a lot longer than initially planned and there were some scary moments when we thought that the challenge of a re-factor of such a large codebase was too much - refer back to the wall of errors when we first ran TypeScript.
```

Comment:
“way;” (comma splice)

Last sentence:
The work done took a lot longer than initially planned, and there were some scary moments when we thought that the challenge of a re-factor of such a large codebase was too much: refer back to the wall of errors when we first ran TypeScript.